src/main/scala/dpla/ingestion3/mappers/providers/Extractor.scala

```diff
@@ -9,4 +10,14 @@ import dpla.ingestion3.model.{DplaMapData, EdmAgent}
 trait Extractor {
   def build(rawData: String): DplaMapData
   def agent: EdmAgent
-}
+  /**
+    * Generates an MD5 checksum from the given providerId
+    * Implemented per provider
+    *
+    * DigestUtils.md5Hex(id)
+    *
+    * @return MD5 hash value
+    * @throws NotImplemented
+    */
+  def mintItemId(rawData: Any): String
```

Comment:
That might be, but I think having the method take Any opens us up to a different set of problems as well. In theory, you could stick anything in there (database connection? file?) and not get any help in knowing that you did the wrong thing.

I think we should use the template pattern here: https://en.wikipedia.org/wiki/Template_method_pattern

Have methods in Extractor that can return the ID or the URI, and have an abstract method that knows how to return the ID?